### PR TITLE
Fix issue related with ordering of httppolicysets of VS having same host with different Paths for Openshift route, K8 ingress

### PR DESCRIPTION
**Pull request by akshayhavile**
As part of this PR, httppolicysets of same VS are sorted in ascending order of path.
Changes has been done for SNI and EVH models.

### CODE REVIEW

**Comment by akshayhavile** [comment]
build ako

File: internal/rest/avi_obj_vs.go [hunk 1]
```diff
@@ -255,6 +256,11 @@ func (rest *RestOperations) AviVsSniBuild(vs_meta *nodes.AviVsNode, rest_method
 
 		var httpPolicyCollection []*avimodels.HTTPPolicies
 		internalPolicyIndexBuffer := int32(11)
+
+		sort.Slice(vs_meta.HttpPolicyRefs, func(i, j int) bool {
```
Comment (sudswasavi) [inline internal/rest/avi_obj_vs.go:260]: For this type of sorting, we aren't considering the noPath condition. but with EVH we are doing that. If you had a host without any path - will path[0] have a "/"?

Comment (akshayhavile) [reply internal/rest/avi_obj_vs.go]: @sudswasavi : Thanks. Yes. I missed that. With latest changes fixed ordering of no path and with path httpps in VS

**Review by hemantavi — COMMENTED** [review]
Both functions `AviVsChildEvhBuild` and `AviVsSniBuild` are using the same methodology to figure out the index and assign the policies. Can't we take the common stuff out into a function and use that instead? Because, if something had to be changed later, we would have to do at multiple places, instead of just one. @akshayhavile

**Review by sudswasavi — APPROVED** [review]

**Review by chauhanshubham — COMMENTED** [review]
We would need to reverse sort during checksum calculation in layer 2 in order to force a PUT call while publishing in the REST layer
cc: @sudswasavi @monotosh-avi

**Comment by sudswasavi** [comment]
Hi @chauhanshubham I spoke with @akshayhavile . Your concern is a valid one but we plan to address it in a subsequent PR. Currently we create a httppolicyset object per host/path combination which is sub-optimal. We would create a single httppolicyset object per host and then use a sorting function to sort the paths in descending order in layer 2 and publish the model to Layer 3. This will get addressed in the next PR. Hence can we pls merge this for now? cc @hemantavi

**Review by hemantavi — APPROVED** [review]

**Pull request merged**